pkg/gatewayapi/supportedfeatures.go

```diff
@@ -0,0 +1,51 @@
+package gatewayapi
+
+import (
+	"errors"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/gateway-api/pkg/features"
+
+	"github.com/kong/gateway-operator/pkg/consts"
+)
+
+var (
+	traditionalCompatibleRouterSupportedFeatures = commonSupportedFeatures.Clone().Insert(
+	// add here the traditional compatible router specific features
+	)
+
+	expressionsRouterSupportedFeatures = commonSupportedFeatures.Clone().Insert(
+		// extended
+		features.SupportHTTPRouteMethodMatching,
+		features.SupportHTTPRouteQueryParamMatching,
+	)
+)
+
+var (
+	commonSupportedFeatures = sets.New(
+		// core features
+		features.SupportHTTPRoute,
+		features.SupportGateway,
+		features.SupportReferenceGrant,
+
+		// Gateway extended
+		features.SupportGatewayPort8080,
+
+		// HTTPRoute extended
+		features.SupportHTTPRouteResponseHeaderModification,
+		features.SupportHTTPRoutePathRewrite,
+		features.SupportHTTPRouteHostRewrite,
+	)
+)
+
+// GetSupportedFeatures returns the supported features for the given router type.
+func GetSupportedFeatures(routerType consts.RouterFlavor) (sets.Set[features.FeatureName], error) {
+	switch routerType {
+	case consts.RouterFlavorTraditionalCompatible:
+		return traditionalCompatibleRouterSupportedFeatures, nil
+	case consts.RouterFlavorExpressions:
+		return expressionsRouterSupportedFeatures, nil
+	default:
+		return nil, errors.New("unsupported router type")
+	}
+}
```

Comment:
Ditto about https://github.com/Kong/gateway-operator/blob/229786e24a261542b4e171886323d0cd97394f13/test/conformance/conformance_test.go#L82-L93

Reply:
Ditto

Reply:
🤦 Missed it, sorry. 

👍